ai/shared-concepts.md

```diff
@@ -65,13 +63,66 @@ pub struct MyContext {
 }
 ```
 
-Construction should be async:
+**Required Fields:**
+- `#[config] pub env: BlueprintEnvironment` - Provides access to configuration, keystore, data directories - this is always required
+- Other fields should contain the shared clients and state needed by job handlers - they will differ per Blueprint
+
+**Required Traits:**
+- `#[derive(Clone)]` - Contexts must be cloneable for job handlers - this is always required
+- Chain-specific traits as needed:
+  - `TangleClientContext` - For Tangle chain interactions - this should always be used for Tangle Blueprints
+  - `ServicesContext` - For service registry access - this should always be used for all Blueprints
+  - `KeystoreContext` - For direct keystore access - this should always be used for all Blueprints
```

Comment:
These aren't strictly required. Shouldn't encourage the use of derives that aren't necessary